spark/src/main/scala/io/delta/sharing/spark/DeltaSharingSource.scala

```diff
@@ -240,48 +277,90 @@ case class DeltaSharingSource(
       isStartingVersion: Boolean,
       currentLatestVersion: Long): Unit = {
     lastQueryTableTimestamp = System.currentTimeMillis()
+    minUrlExpirationTimestamp = -1
     if (isStartingVersion) {
       // If isStartingVersion is true, it means to fetch the snapshot at the fromVersion, which may
       // include table changes from previous versions.
       val tableFiles = deltaLog.client.getFiles(
         deltaLog.table, Nil, None, Some(fromVersion), None, None
       )
       latestRefreshFunc = () => {
-        deltaLog.client.getFiles(
+        val files = deltaLog.client.getFiles(
           deltaLog.table, Nil, None, Some(fromVersion), None, None
-        ).files.map { f =>
+        ).files
+        var minUrlExpiration: Long = -1L
+        val idToUrl = files.map { f =>
+          if (f.expirationTimestamp != null) {
+            if (minUrlExpiration == -1L) {
+              minUrlExpiration = f.expirationTimestamp
+            } else {
+              minUrlExpiration.min(f.expirationTimestamp)
+            }
+          }
           f.id -> f.url
         }.toMap
+
+        refreshSortedFetchedFiles(idToUrl)
+
+        (idToUrl, minUrlExpiration)
```

Comment:
If we also set the minUrlExpirationTimestamp as part of  refreshSortedFetchedFiles, can we avoid the complexity of always checking for -1?

Reply:
we avoid checking for -1 by setting it as Long.MaxValue.

What do you mean by set the minUrlExpieraetionTimestamp as part of refrshSortedFetchedFiles?

Reply:
It will be desirable if we update the sortedFetchedFiles and minUrlExpirationTime in one function together.
It would be easy to apply some config (for example, to enable/disable this) to that one place if needed.